ui/src/frontend/overview_timeline_panel.ts

```diff
@@ -14,216 +14,197 @@
-          this.onDragEnd.bind(this));
+
+    onremove({ dom }: m.CVnodeDOM) {
+        (dom as HTMLElement).removeEventListener('mousemove', this.boundOnMouseMove);
+    }
+
+    view() {
+        return m('.overview-timeline');
     }
-  }
-
-  oncreate(vnode: m.CVnodeDOM) {
-    this.onupdate(vnode);
-    (vnode.dom as HTMLElement)
-        .addEventListener('mousemove', this.boundOnMouseMove);
-  }
-
-  onremove({dom}: m.CVnodeDOM) {
-    (dom as HTMLElement)
-        .removeEventListener('mousemove', this.boundOnMouseMove);
-  }
-
-  view() {
-    return m('.overview-timeline');
-  }
-
-  renderCanvas(ctx: CanvasRenderingContext2D, size: PanelSize) {
-    if (this.width === undefined) return;
-    if (this.traceTime === undefined) return;
-    if (this.timeScale === undefined) return;
-    const headerHeight = 20;
-    const tracksHeight = size.height - headerHeight;
-
-    if (size.width > TRACK_SHELL_WIDTH && this.traceTime.duration > 0n) {
-      const maxMajorTicks = getMaxMajorTicks(this.width - TRACK_SHELL_WIDTH);
-      const tickGen = new TickGenerator(
-          this.traceTime, maxMajorTicks, globals.state.traceTime.start);
-
-      // Draw time labels on the top header.
-      ctx.font = '10px Roboto Condensed';
-      ctx.fillStyle = '#999';
-      for (const {type, time} of tickGen) {
-        const xPos = Math.floor(this.timeScale.tpTimeToPx(time));
-        if (xPos <= 0) continue;
-        if (xPos > this.width) break;
-        if (type === TickType.MAJOR) {
-          ctx.fillRect(xPos - 1, 0, 1, headerHeight - 5);
-          const sec = tpTimeToSeconds(time - globals.state.traceTime.start);
-          ctx.fillText(sec.toFixed(tickGen.digits) + ' s', xPos + 5, 18);
-        } else if (type == TickType.MEDIUM) {
-          ctx.fillRect(xPos - 1, 0, 1, 8);
-        } else if (type == TickType.MINOR) {
-          ctx.fillRect(xPos - 1, 0, 1, 5);
+
+    renderCanvas(ctx: CanvasRenderingContext2D, size: PanelSize) {
+        if (this.width === undefined) return;
+        if (this.traceTime === undefined) return;
+        if (this.timeScale === undefined) return;
+        const headerHeight = 20;
+        const tracksHeight = size.height - headerHeight;
+
+        if (size.width > TRACK_SHELL_WIDTH && this.traceTime.duration > 0n) {
+            const maxMajorTicks = getMaxMajorTicks(this.width - TRACK_SHELL_WIDTH);
+            const tickGen = new TickGenerator(this.traceTime, maxMajorTicks, globals.state.traceTime.start);
+
+            // Set a background of color --perfetto-overview-background
+            const perfettoOverviewBackground = getComputedStyle(document.documentElement).getPropertyValue(
+                '--perfetto-overview-background'
+            );
+            // getPropertyValue returns output of" "actualColor" "
+            // This cuts it down to "actualColor"
+            const style = perfettoOverviewBackground.substring(2, perfettoOverviewBackground.length - 1) || 'white';
```

Comment:
I'd be more comfortable with a simple `trim()` invocation, because this depends on how the CSS is authored. For example, in my first test there wasn't a trailing space in the value of `perfettoOverviewBackground`.

Reply:
I think I have fixed this with my most recent commit